.github/workflows/ci-go.yml

```diff
@@ -28,6 +28,86 @@ jobs:
       - uses: golangci/golangci-lint-action@v3.1.0
         with:
           skip-go-installation: true
+  terraform-provider-corner-tfprotov5:
+    defaults:
+      run:
+        working-directory: terraform-provider-corner
+    name: tfprotov5 (${{ matrix.os}}, Terraform ${{ matrix.terraform}})
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/checkout@v3
+        with:
+          path: terraform-provider-corner
+          repository: hashicorp/terraform-provider-corner
+      - id: go-version
+        # Reference: https://github.com/actions/setup-go/issues/23
+        run: echo "::set-output name=version::$(cat ./.go-version)"
+        working-directory: .
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ steps.go-version.outputs.version }}
+      - uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: ${{ matrix.terraform }}
+          terraform_wrapper: false
+      - run: go mod edit -replace github.com/hashicorp/terraform-plugin-go=../
+      - run: go mod tidy
+      - run: go test -v ./internal/protocolprovider
+      - run: go test -v ./internal/sdkv2provider
+        env:
+          TF_ACC: '1'
+      - run: go test -v ./internal/tf5muxprovider
+      - run: go test -v ./internal/tf6to5provider
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-latest, windows-latest, ubuntu-latest]
+        terraform:
+          - '0.12.*'
+          - '0.13.*'
+          - '0.14.*'
+          - '0.15.*'
+          - '1.0.*'
+          - '1.1.*'
+  terraform-provider-corner-tfprotov6:
+    defaults:
+      run:
+        working-directory: terraform-provider-corner
+    name: tfprotov6 (${{ matrix.os}}, Terraform ${{ matrix.terraform}})
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/checkout@v3
+        with:
+          path: terraform-provider-corner
+          repository: hashicorp/terraform-provider-corner
+      - id: go-version
+        # Reference: https://github.com/actions/setup-go/issues/23
+        run: echo "::set-output name=version::$(cat ./.go-version)"
+        working-directory: .
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ steps.go-version.outputs.version }}
+      - uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: ${{ matrix.terraform }}
+          terraform_wrapper: false
+      - run: go mod edit -replace github.com/hashicorp/terraform-plugin-go=../
+      - run: go mod tidy
+      - run: go test -v ./internal/frameworkprovider
+      - run: go test -v ./internal/protocolv6provider
+      - # Terraform CLI 1.1.5+ is required for upgrading sdk/v2
+        if: matrix.terraform != '1.0.*'
+        run: go test -v ./internal/tf5to6provider
```

Comment:
Correct! There was an accidental omission in the version 6 protocol that's required for SDK/v2 usage. So instead of supporting 1.0.3+ in that scenario, it's 1.1.5+. This was acceptable as the recommendation (due to 1.x compatibility promises) is to just jump to the latest 1.x when upgrading from 0.14/0.15 (instead of say 1.0.11 in between).